x-pack/plugins/infra/server/lib/alerting/metric_threshold/lib/create_timerange.test.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createTimerange } from './create_timerange';
+import { Aggregators } from '../types';
+import moment from 'moment';
+
+describe('createTimerange(interval, aggType, timeframe)', () => {
+  describe('without timeframe', () => {
+    describe('Basic Metric Aggs', () => {
+      it('should return a second range for last 1 second', () => {
+        const subject = createTimerange(1000, Aggregators.COUNT);
+        expect(subject.end - subject.start).toEqual(1000);
```

Comment:
I left `timeframe` optional because it's optional for the parent function `getMetric` and I wanted to encapsulate all the "time frame" logic in one place. Otherwise we would need to do something like
```
const calculatedTimeframe = createTimerange(intervalAsMS, aggType, timeframe || { end: (Date.now()).valueOf() }); 
```
instead of 
```
const calculatedTimeframe = createTimerange(intervalAsMS, aggType, timeframe);
```
which I thought looked cleaner.

Reply:
I figured the former would be easier to test and more "self explanatory" as to what is happening.  Here you can only test the interval of the dates and not what the dates are.  But I'm fine either way.